home-exercises/src/main/java/extrapractice/bta/LetterCounter.java

```diff
@@ -0,0 +1,57 @@
+package extrapractice.bta;
+
+import java.io.*;
+import java.util.HashMap;
+import java.util.stream.Collectors;
+
+public class LetterCounter {
+    static File f = new File("resources/text_file_1000_chars.txt");
+
+    public static void main(String[] args) {
+        HashMap<String, Integer> letterHash = readChars(f);
+        writeFile(letterHash);
+    }
+
+
+    public static <T, V> String mapToString(HashMap<T, V> map) {
+        return map.keySet().stream().map(key -> key + "=" + map.get(key)).collect(Collectors.joining(", ", "{", "}"));
+    }
+
+    public static HashMap<String, Integer> readChars(File f) {
+        BufferedReader reader;
+        String line;
+        HashMap<String, Integer> letterHash = new HashMap<>();
+        try {
+            reader = new BufferedReader(new FileReader(f));
+            while ((line = reader.readLine()) != null) {
+                String[] split = line.replaceAll(" ", "").replaceAll("[^a-zA-Z]", "").toUpperCase().split("");
+                if (line.equals("")) {
+                    continue;
+                }
+                for (String s : split) {
+                    if (!letterHash.containsKey(s)) {
+                        letterHash.put(s, 1);
+                    } else {
+                        letterHash.put(s, letterHash.get(s) + 1);
+                    }
+                }
+            }
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        return letterHash;
+    }
+
+    public static void writeFile(HashMap<String, Integer> letterHash) {
```

Comment:
Ar šo uzmanīgi - uzdevumā nebija prasīts, ka rezultāts ir jāraksta failā. Pats arī komentārā mini, ka līdz ar to ir problēmas ar testiem. Ir labi, ja papildini kodu ar lietām, kuras Tu zini, bet šobrīd faktiski 1/3 no koda, kas te ir uzrakstīts netika prasīts. Tas arī dod vairāk iespējas, kur piekasīties. Tāpēc mans ieteikums ir pilnībā izpildīt prasības un tad domāt, ko var pielikt klāt, bet šobrīd es uzrakstītjis writeToFile, bet testa nav.